mat/format.go

```diff
@@ -87,6 +87,52 @@ func (f formatter) Format(fs fmt.State, c rune) {
 	f.format(f.matrix, f.prefix, f.margin, f.dot, f.squeeze, fs, c)
 }
 
+// formattable holds a matrix and provides methods for formatting values.
+type formattable interface {
+	tabular
+
+	// NewValueFormatter returns a new, zero-valued fmt.Formatter
+	NewValueFormatter(options ...vFormatOption) fmt.Formatter
+
+	// RevalueFormatter sets the value of a formatter to element i, j.
+	// A use is to recycle an existing value formatter for efficiency.
+	RevalueFormatter(f fmt.Formatter, i, j int) fmt.Formatter
+
+	// ValueFormatter returns a new fmt.Formatter for element i, j
+	ValueFormatter(i, j int, options ...vFormatOption) fmt.Formatter
+
+	// Matrix returns the source matrix
+	Mat() tabular
+}
+
+// formattableMatrix is a formattable for floating point-valued matrices, those
+// that satisfy the Matrix interface.
+type formattableMatrix struct{ mat Matrix }
+
+var _ formattable = (*formattableMatrix)(nil)
```

Comment:
This should not check against the point since the methods are on the concrete type.